### ISSUE TITLE
Emojis in commit message

### ISSUE THREAD

**Issue by GoOz** [cozy/cozy-guidelines#48]
Some of us like to add (related) emojis to their commit messages, some of us don't.

What do we do?

Can we use emojis? And if so, do we set a rule to its location in the message? In the beginning just before the type or at the end of the message?

Discuss or vote!

🎉 At the beginning
❤️ At the end
👎 Against emojis
😕 I don't care

**Comment by kosssi** [comment]
👍 to the end of message, because it's not important

**Comment by enguerran** [comment]
Useful git commit message is short and to stay concise a good commit message avoid useless characters (think twitter in its earlier stage or in SMS protocol).

If you think an emoji is mandatory, then it should be placed in the `Subject` as defined in the current guidelines: `feat: :hammer_and_wrench: Informative subject` or a more concise version with the unicode character: `feat: 🛠 Informative subject with more characters`.

But git commit history ain't [a bullet journal](https://www.wikihow.com/Bullet-Journal), it is useful for humans and machines to understand what is going on on the project. 

There is plainty of git client, like github, official git CLI or gitk, that will display messages in one or another format (only the first 50 characters, `:text_unicode:` replaced by the corresponding emoji, etc.). To ensure every body could read in a very effective way the git history, we have to keep only very useful characters in the Subject.

You can add more details in the Body of the commit message (after the subject, insert a blank line and be as long as you need to explain, illustrate, etc. the change you are commiting).

For instance, I created a blank git project with different kind of message:
https://github.com/enguerran/emojiincommit/commits/master

Take a look at the git commit messages in github:
![image](https://user-images.githubusercontent.com/701648/42325111-81a2a0a2-8065-11e8-9214-2b34c1f6429b.png)

With `git log` in my own terminal:
![image](https://user-images.githubusercontent.com/701648/42325141-962b6b62-8065-11e8-93d3-6fff47f36ac3.png)

In gitk:
![image](https://user-images.githubusercontent.com/701648/42325164-a6f9e0d6-8065-11e8-9267-1a27b8607409.png)

A lot of people recommend to stay to 50 characters in the subject (see Linux kernel repo):
- https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
- https://stackoverflow.com/questions/2290016/git-commit-messages-50-72-formatting
- https://udacity.github.io/git-styleguide/
- https://git.wiki.kernel.org/index.php/CommitMessageConventions
- http://365git.tumblr.com/post/3308646748/writing-git-commit-messages
- https://commit.style/
- https://chris.beams.io/posts/git-commit/

**Comment by GoOz** [comment]
Correct me if I'm wrong but an emoji as an UTF-8 character is 1 character is the count of 50, isn't it?
I personally think that an Emoji is a good way to display a more or less long information with only one character, so I don't see it as useless.

**Comment by enguerran** [comment]
Yes, it is one character if you use the Unicode version, but the length is not defined if you use the `:text_emoji:` version. That is illustrated in the picture: prefer the Unicode version.

The important is to be clear for other people. If every body understand what `fix: Transplant a new ♥️` means, I see nothing to stop using an emoji instead of a word (`fix: Transplant a new heart`).

**Comment by ptbrowne** [comment]
I am reading the commits of cozy-bar that have emojis in front. I find it very hard to parse. Since when scanning you are watching for fix/feat/chore etc... The emojis in front just add noise. I would be in favor of banning this usage :)

**Comment by GoOz** [comment]
> I would be in favor of banning this usage :)

of emojis at the beginning of the commit message or emojis at all ?

The thing is every argument I read is only about personal perception. I personally like emojis, I think that an emoji is actually far more easy to parse with your eyes than reading a word. Well, this works only if we all use the same "rules" of use. But I do understand that some of us could be insensitive to emojis.

As far I see it, nobody's right, nobody's wrong. But is there a way to find a middle ground here, so everybody can be happy? Maybe by writing emojis at the end of the commit for example.

**Comment by drazik** [comment]
Yep, the middle ground would be to put emojis at the end of the commit message. This way automatic releases don't need a specific parser, those who like the emoji's visual information have it, and those who don't like emojis don't have it under their eyes directly.

**Comment by ptbrowne** [comment]
Seems OK to me, GoOz would you be OK to do the PR ?

**Comment by GoOz** [comment]
Sure
